src/pages/ProductList/ProductList.js

```diff
@@ -1,5 +1,37 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Route, Routes } from 'react-router-dom';
+import ProductListNav from './ProductListNav/ProductListNav';
+import ProductsMain from './ProductsMain/ProductsMain';
 
-export default function ProductList() {
-  return <div>hi</div>;
-}
+const ProductList = () => {
+  const [isProductLoading, setIsProductLoading] = useState(false);
+  const [productsList, setProductsList] = useState([]);
+
+  const fetchData = async () => {
+    setIsProductLoading(true);
+    const data = await fetch('/data/product_data.json');
+    const res = await data.json();
+    setProductsList(res);
+  };
+
+  useEffect(() => {
+    fetchData();
+    setIsProductLoading(false);
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    (async () => {
         await fetchData()
         setIsProductLoading(false);
    })()
  }, []);
```
원하는 동작을 시행하려면 이렇게 되어야 할 것 같네요!
async await, 비동기에 대해서 좀 더 공부하시면 좋을 것 같네요!! 지금 당장은 이해하기 어려운 내용이라서 완벽히 이해하려 하기보다는 대략적인 개념만 잡고 실제로 개발 공부 점점 하시면서 조금씩 깊게 공부해보시면 좋을 것 같습니다!

Reply:
setIsProductLoading(false)를 fetchData 안으로 넣어주는 방향으로 해결했습니다!

Reply:
이 방법은 fetchData 함수 안에서 loading state를 관리하는 것 까지 진행을 해버리기에 함수가 기능별로 분리가 잘 안됬다라는 측면에서 리팩토링할 여지가 남아있긴 합니다! 다만 여러가지 방법을 잘 찾아보고 고민해보고 적용해신 부분은 너무 좋네요! 추후 리팩토링 진행하시죠 머지하겠습니다

Reply:
참고하고 추후에 꼭 변경하도록 하겠습니다!